src/pkg/repository/repository.go

```diff
@@ -345,6 +266,68 @@ func (r repository) Counter() *count.Bus {
 	return r.counter
 }
 
+func (r *repository) addContextInfo(ctx context.Context) context.Context {
+	return clues.Add(
+		ctx,
+		"acct_provider", r.Account.Provider.String(),
+		"acct_id", clues.Hide(r.Account.ID()),
+		"storage_provider", r.Storage.Provider.String())
+}
+
+func (r *repository) handleRecovery(ctx context.Context, err *error, label string) {
+	if crErr := crash.Recovery(ctx, recover(), label); crErr != nil {
+		*err = crErr
+	}
+}
```

Comment:
Can you double check that this correctly produces a trace back to the calling func, and not to this func?  It'll be a regression if all callers of this func report line 278 as the "place where the panic was handled"

In general, I don't think the abstraction earns us much in this case.  We would also be fine keeping the code how it was.

```suggestion
func (r *repository) handleRecovery(ctx context.Context, err *error, label string) {
	if crErr := crash.Recovery(ctx, recover(), label); crErr != nil {
		*err = crErr
	}
}
```